### PR TITLE
fix: eslint issues in report, transaction and other services, continued from base branch

### PR DESCRIPTION
**Pull request by suyashpatil78**
[workerB](https://workerb.linearb.io/v2/badge/collaboration-page?magicLinkId=8dR2WoN)
### <samp>🤖 Generated by Copilot at fe26b91</samp>

This pull request improves the type safety and readability of the code related to expenses, reports, and statistics features by using more specific and consistent types, importing types from models, and updating mock data. It also adds new mock data, models, and methods for the outbox queue, delete popover, report permissions, and report state mapping features.

### <samp>🤖 Generated by Copilot at fe26b91</samp>

> _Sing, O Muse, of the valiant code review_
> _That refined the types of expenses and reports_
> _And made the mock data more consistent and true_
> _With the help of interfaces and imports._

### <samp>🤖 Generated by Copilot at fe26b91</samp>

*  Import `Observable` type from `rxjs` and change `mockDeleteMethod` to return `Observable<void>` in mock data ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-177336f82d99802f4cf3755127f7be1916fd16827397cc57ef02615d20e3e08cL1-R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-177336f82d99802f4cf3755127f7be1916fd16827397cc57ef02615d20e3e08cL227-R227))
* Add `outboxQueueData1` mock data for outbox queue feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-41332ea98266882422f7264d802c5ca1bfa02268675ee3ce209e277c3e06f381R1-R20))
* Import `Datum` type from `../models/v2/stats-response.model` and add `transactionDatum1`, `transactionDatum2`, and `transactionDatum3` mock data for transaction stats feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-bada017d42923e2a5ca91aa2de7a96e69151d5ca42e5720f11998b2158ce5cebL1-R1), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-bada017d42923e2a5ca91aa2de7a96e69151d5ca42e5720f11998b2158ce5cebR319-R416))
* Import `optionsData15` and `optionsData33` mock data from `./merge-expenses-options-data.data` and use them in `editExpTxn2` to `editExpTxn5` mock transactions ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-b8ffe3caa9c9df1506a4592f97c8d56122e9cf79615e9b20dbeceab72a329899L1-R2), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-b8ffe3caa9c9df1506a4592f97c8d56122e9cf79615e9b20dbeceab72a329899L3283-R3283), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-b8ffe3caa9c9df1506a4592f97c8d56122e9cf79615e9b20dbeceab72a329899L3392-R3392), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-b8ffe3caa9c9df1506a4592f97c8d56122e9cf79615e9b20dbeceab72a329899L3501-R3501), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-b8ffe3caa9c9df1506a4592f97c8d56122e9cf79615e9b20dbeceab72a329899L3611-R3611))
* Import `optionsData15` and `optionsData33` mock data from `./merge-expenses-options-data.data` and use them in `expectedUnflattendedTxnData2` and `expectedUnflattendedTxnData3` mock unflattened transactions ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-21ae928ce640ad62cfada5fee7bf0bbc490300ab04a25f8309a9878e21b2f1cfR2), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-21ae928ce640ad62cfada5fee7bf0bbc490300ab04a25f8309a9878e21b2f1cfL565-R566), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-21ae928ce640ad62cfada5fee7bf0bbc490300ab04a25f8309a9878e21b2f1cfL695-R696))
* Add `DeletePopoverParams` and `ExpenseDeletePopoverParams` interfaces for delete popover component props and expense delete method ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-a628e4648637596241de38917d9d17f862cb2a2609752e5bfae6be20d5628d36R1-R29))
* Change `Expense` interface to use more specific types for various fields instead of `any` ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L12-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L30-R33), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L40-R53), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L59-R82), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L81-R190), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-5d1df581fe858623a0a33d1c6f571e5561d7fbf25dbfdabe13f94fedc0964845L213-R220))
* Change `FilterQueryParams` interface to allow `or` to be either a string or an array of strings, and to add `tx_state` as an optional field ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-1bc8d5a29e9cadfffd34844853db0be9b514f8b40f63fd7c304661fb5181cc67L2-R7))
* Change `GetExpensesQueryParamsWithFilters` type to use `FilterQueryParams` instead of `GetExpensesQueryParams` for the `queryParams` field ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-d6e7d60d5f53d72363eeb840d15c75289c65c37f5a977974cb105614ad0cddd8L1-R5))
* Change `GetExpensesQueryParams` type to allow `or` to be either a string or an array of strings ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-e5aa1f42c315ea18e79c49146dfbe0b7d45ee436a1e5769fe01d23b6f6468301L2-R2))
* Add `OutboxQueue` interface for outbox queue feature for offline expense creation ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-d46acf58284cc720984d054a82c811e1beb9de1255f500651d22231d6802b213R1-R15))
* Change `ParsedResponse` interface to use more specific types for various fields instead of `any` ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-7df2ab990a1f4a071016091abe94c98c87763f139dfa46a90cb516fc83994b28L1-R10))
* Add `ReportPermission` interface for report permissions feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-34d53c40427491e5c6da0f7a98d6b6598a69b57096eb06f2b87ba8fc5c0eb1aaR1-R6))
* Add `ReportStateMap` interface for report state mapping feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-57f2f6ea5654efad0ff85cf29c2ece4ce9c2144a1c3bc8fb97357826c3f1fc68R1-R26))
* Add `ReportStatsResponse` interface for report stats response feature ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-19507ee4f448455a1083b440568117da5cc9b863229c18b083dd4a69294ade11R1-R6))
* Import `Destination` type from `../destination.model` and use `Destination[]` instead of `any` for the `locations` field of the `Transaction` interface ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-db96e89a4671a6577872d9c95a3722abc4d79e6b3f8443ec4f74cedb81915dbaR1), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-db96e89a4671a6577872d9c95a3722abc4d79e6b3f8443ec4f74cedb81915dbaL36-R37))
* Change `StatsResponse` class to use `Datum` as the return type for the `getDatum` method instead of `any` ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-9a199c1818c647d83e2bb24925f2ca1db1e5f57a2cd19ac3389ab7bbf3ffe1c6L47-R47))
* Change `downloadSummaryPdfUrl` method in `ReportService` class to return the `report_url` property of the response instead of the whole response object ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-609c1fb70188be62ef86ff68c50f64df0f710b41ce36c8b1109fc2034ee03524L721-R724), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L449-R457))
* Import `Datum`, `ReportStateMap`, and `ReportPermission` types from respective models and use them in various methods of `ReportService` class, and change the return types and parameters of some methods to use more specific types instead of generics or `any` ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L15-R15), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9R33-R34), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L129-R133), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L144-R148), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L166-R168), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L173-R175), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L180-R184), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L189-R191), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L196-R198), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L204-R206), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L211-R213), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L218-R220), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L256-R261), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L274-R277), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L381-R384), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L497-R499), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L511-R513), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L517-R519), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L555-R561), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-12dc98777c493c39a3e7aec29b439cc7863ddfb002bf1e52052c8aee485c63c9L570-R572))
* Import `Datum` type from `../models/v2/stats-response.model` and use it in various methods of `TasksService` class, and change the return types and parameters of some methods to use `Datum[]` instead of `StatsResponse` ([link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL21-R22), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL427-R427), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL440-R435), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL446-R449), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL460-R455), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL468-R463), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL474-R482), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL498-R492), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL561-R555), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL567-R561), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL573-R567), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL586-R580), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL786-R780), [link](https://github.com/fylein/fyle-mobile-app/pull/2218/files?diff=unified&w=0#diff-32ef5239e59eae00c95f2691e409899dbc8aa366b379a6f762fa640d199b7e0aL798-R800))

## Clickup
app.clickup.com

## Code Coverage
Please add code coverage here

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/services/report.service.ts [hunk 1]
```diff
@@ -126,9 +128,9 @@ export class ReportService {
   @CacheBuster({
     cacheBusterNotifier: reportsCacheBuster$,
   })
-  addTransactions<T>(rptId: string, txnIds: string[]): Observable<T> {
+  addTransactions(rptId: string, txnIds: string[]): Observable<Expense> {
```
Comment (Julias0) [inline src/app/core/services/report.service.ts:131]: Are we sure this endpoint returns a Expense in the response?

File: src/app/core/services/report.service.ts [hunk 2]
```diff
@@ -141,9 +143,9 @@ export class ReportService {
   @CacheBuster({
     cacheBusterNotifier: reportsCacheBuster$,
   })
-  createDraft<T>(report: ReportPurpose): Observable<T> {
+  createDraft(report: ReportPurpose): Observable<ReportV1> {
```
Comment (Julias0) [inline src/app/core/services/report.service.ts:146]: same as above - can you paste a sample response

File: src/app/core/services/report.service.ts [hunk 3]
```diff
@@ -163,59 +165,59 @@ export class ReportService {
   @CacheBuster({
     cacheBusterNotifier: reportsCacheBuster$,
   })
-  removeTransaction<T>(rptId: string, txnId: string, comment?: string[]): Observable<T> {
+  removeTransaction(rptId: string, txnId: string, comment?: string[]): Observable<Expense> {
```
Comment (Julias0) [inline src/app/core/services/report.service.ts:168]: Can you paste a sample of the response of the API call

File: src/app/core/models/delete-popover-params.model.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  backdropDismiss: boolean;
+  componentProps: {
+    header: string;
+    body: string;
+    infoMessage: string;
+    deleteMethod: () => Observable<void>;
+  };
+}
+
+export interface ExpenseDeletePopoverParams {
```
Comment (Julias0) [inline src/app/core/models/delete-popover-params.model.ts:17]: move to separate file

File: src/app/core/services/report.service.ts (hunk 2 above)
Comment (suyashpatil78) [reply src/app/core/services/report.service.ts:146]: {
    "created_at": "2023-08-01T06:27:55.169Z",
    "updated_at": "2023-08-01T06:27:55.169Z",
    "last_updated_by": {
        "user_id": "usvKA4X8Ugcr",
        "org_user_id": "oudmNnyXjIgs",
        "org_id": "ork30mHcFSW3",
        "roles": [
            "FYLER",
            "OWNER",
            "ADMIN",
            "APPROVER",
            "HOD"
        ],
        "scopes": [],
        "allowed_CIDRs": [],
        "cluster_domain": "\"https://staging.fyle.tech\"",
        "proxy_org_user_id": null,
        "tpa_id": null,
        "tpa_name": null,
        "name": "oudmNnyXjIgs"
    },
    "id": "rpcywYj3t0VW",
    "org_user_id": "oudmNnyXjIgs",
    "purpose": "travel",
    "currency": "INR",
    "creator_id": "oudmNnyXjIgs",
    "amount": 0,
    "tax": null,
    "status_id": null,
    "num_transactions": 0,
    "tally_export_id": null,
    "state": "DRAFT",
    "source": "MOBILE",
    "reimbursement_id": null,
    "approved_at": null,
    "reimbursed_at": null,
    "submitted_at": null,
    "settlement_id": null,
    "verification_state": null,
    "trip_request_id": null,
    "physical_bill": false,
    "exported": null,
    "manual_flag": null,
    "policy_flag": null,
    "claim_number": "C/2023/08/R/2",
    "physical_bill_at": null,
    "from_dt": null,
    "to_dt": null,
    "location1": null,
    "location2": null,
    "location3": null,
    "location4": null,
    "location5": null,
    "type": "EXPENSE"
}

Comment (suyashpatil78) [reply src/app/core/services/report.service.ts:146]: This is `ReportV1` interface

File: src/app/core/models/delete-popover-params.model.ts (hunk 4 above)
Comment (suyashpatil78) [reply src/app/core/models/delete-popover-params.model.ts]: @Julias0 made all the req changes. Please check

File: src/app/core/services/report.service.ts (hunk 3 above)
Comment (suyashpatil78) [reply src/app/core/services/report.service.ts]: It returns `void`

File: src/app/core/services/report.service.ts [hunk 5]
```diff
@@ -446,13 +448,13 @@ export class ReportService {
...
-  delete<T>(rptId: string): Observable<T> {
+  delete(rptId: string): Observable<void> {
     return this.apiService
-      .delete<T>('/reports/' + rptId)
+      .delete<void>('/reports/' + rptId)
       .pipe(switchMap((res) => this.clearTransactionCache().pipe(map(() => res))));
   }
 
-  downloadSummaryPdfUrl<T>(data: { report_ids: string[]; email: string }): Observable<T> {
+  downloadSummaryPdfUrl(data: { report_ids: string[]; email: string }): Observable<{ report_url: string }> {
```
Comment (suyashpatil78) [inline src/app/core/services/report.service.ts:457]: `{
    "report_url": "mockURL"
}`

File: src/app/core/services/report.service.ts [hunk 6]
```diff
@@ -203,21 +203,21 @@ export class ReportService {
...
-  approve(rptId: string): Observable<boolean> {
+  approve(rptId: string): Observable<void> {
     return this.apiService.post('/reports/' + rptId + '/approve');
   }
 
   @CacheBuster({
     cacheBusterNotifier: reportsCacheBuster$,
   })
-  addApprover(rptId: string, approverEmail: string, comment: string): Observable<Approver[]> {
+  addApprover(rptId: string, approverEmail: string, comment: string): Observable<void> {
```
Comment (suyashpatil78) [inline src/app/core/services/report.service.ts:220]: Checked this, it returns `void`

File: src/app/core/services/report.service.ts [hunk 7]
```diff
@@ -203,21 +203,21 @@ export class ReportService {
       };
       notify: boolean;
     }
-  ): Observable<boolean> {
+  ): Observable<void> {
```
Comment (suyashpatil78) [inline src/app/core/services/report.service.ts:206]: Returns `void`

File: src/app/core/services/report.service.ts [hunk 8]
```diff
@@ -203,21 +203,21 @@ export class ReportService {
...
     return this.apiService.post('/reports/' + rptId + '/inquire', addStatusPayload);
   }
 
   @CacheBuster({
     cacheBusterNotifier: reportsCacheBuster$,
   })
-  approve(rptId: string): Observable<boolean> {
+  approve(rptId: string): Observable<void> {
```
Comment (suyashpatil78) [inline src/app/core/services/report.service.ts:213]: returns `void`

File: src/app/core/services/transaction.service.ts [hunk 9]
```diff
@@ -717,11 +720,11 @@ export class TransactionService {
     const newQueryParamsCopy = cloneDeep(newQueryParams);
     if (filters.splitExpense) {
       if (filters.splitExpense === 'YES') {
-        newQueryParamsCopy.or.push('(tx_is_split_expense.eq.true)');
+        (newQueryParamsCopy.or as string[]).push('(tx_is_split_expense.eq.true)');
```
Comment (Julias0) [inline src/app/core/services/transaction.service.ts:723]: why do we need `as` here? We should avoid that

**Review by Julias0 — APPROVED** [review]

**Pull request merged**